oauthproxy_test.go

```diff
@@ -1776,10 +1780,75 @@ func TestGetRedirect(t *testing.T) {
 			url:              proxy.ProxyPrefix + "/foo/bar",
 			expectedRedirect: "/",
 		},
+		{
+			name: "proxied request outside of ProxyPrefix redirects to proxiexd URL",
+			url:  "https://oauth.example.com/foo/bar",
+			headers: map[string]string{
+				"X-Forwarded-Proto": "https",
```

Comment:
Thanks for consolidating! We're almost there.  We just need to update the rigor of configuration combinations that these test cases hit now that this function has 3 competing code paths for implemention (`rd`, `X-Forwarded-*`, `X-Auth-Request-Redirect`.

So we need test cases without any of these headers - since that will be the behavior of the use case for a majority of users.

You can likely leave the original test cases that didn't have headers and expand with more cases that have combinations of `X-Forwarded-*` and `X-Auth-Request-Redirect` headers.

We also need a test cases added that has `?rd=<some_uri>` to make sure that takes precedence over any `X-Forwarded-*` headers in all cases.

Reply:
Sure! I'll get it committed in a second.

Reply:
Cool! Thanks again for bearing with me for the last minute tweaks 😅 

Since OAuth redirect logic is always ripe for security bugs, I like to err on the side of over testing all the permutations of cases.